test/mtz-ajax-interceptor_test.html

```diff
@@ -9,34 +9,265 @@
+            const spy = sinon.spy(element.runInterceptors, 'bind');
+            element.ready();
+            expect(spy).calledWith(element);
+          });
+          it('should bind this to __registerInterceptor', () => {
+            const spy = sinon.spy(element.__registerInterceptor, 'bind');
+            element.ready();
+            expect(spy).calledWith(element);
+          });
+          it('should bind this to __unregisterInterceptor', () => {
+            const spy = sinon.spy(element.__unregisterInterceptor, 'bind');
+            element.ready();
+            expect(spy).calledWith(element);
+          });
+        });
+        /* Public Methods */
+        describe('runInterceptors(eventName, event)', () => {
+          beforeEach(() => {
+            element.__unregisterInterceptor = sinon.spy();
+          });
+
+          it('should do nothing when there is no event', () => {
+            element.interceptors = {
+              foo: new Set([{intercept: sinon.spy()}])
+            };
+            element.runInterceptors('foo');
+            expect(Array.from(element.interceptors.foo)[0].intercept.called).to.equal(false);
+          });
+          it('should call intercept(event, eventName) on all interceptors for eventName', () => {
+            const event = new CustomEvent('foo');
+            element.interceptors = {
+              foo: new Set([{intercept: sinon.spy()}, {intercept: sinon.spy()}]),
+              bar: new Set([{intercept: sinon.spy()}])
+            };
+            element.runInterceptors('foo', event);
+            expect(Array.from(element.interceptors.bar)[0].intercept.called).to.equal(false);
+            expect(Array.from(element.interceptors.foo)[0].intercept).calledWithMatch(event, 'foo');
+            expect(Array.from(element.interceptors.foo)[1].intercept).calledWithMatch(event, 'foo');
+          });
+          it('should sort interceptors based on priority', () => {
+            const event = new CustomEvent('foo');
+
+            element.interceptors = {
+              foo: new Set([
+                {
+                  intercept: sinon.spy(),
+                  priority: -1
+                },
+                {
+                  intercept: sinon.spy(),
+                  priority: 1
+                },
+                {
+                  intercept: sinon.spy(),
+                  priority: 0
+                }
+              ]),
+            };
+            element.runInterceptors('foo', event);
+            const interceptors = Array.from(element.interceptors.foo);
+            expect(interceptors[0].intercept).calledWithMatch(event, 'foo');
+            expect(interceptors[1].intercept).calledWithMatch(event, 'foo');
+            expect(interceptors[2].intercept).calledWithMatch(event, 'foo');
+            expect(interceptors[1].intercept).calledBefore(interceptors[2].intercept);
+            expect(interceptors[2].intercept).calledBefore(interceptors[0].intercept);
```

Comment:
Order: 1 => 2 => 0
Checks:
* 1 => 2
* 2 => 0

The only one not explicitly checked is 1 => 0, but that's implied since 2 is there.